ckan/lib/navl/dictization_functions.py

```diff
@@ -219,35 +220,22 @@ def augment_data(data, schema):
 
 
 def convert(converter, key, converted_data, errors, context):
-
-    try:
-        value = converter(converted_data.get(key))
-        converted_data[key] = value
-        return
-    except TypeError as e:
-        # hack to make sure the type error was caused by the wrong
-        # number of arguments given.
-        if converter.__name__ not in str(e):
-            raise
-    except Invalid as e:
-        errors[key].append(e.error)
-        return
-
-    try:
-        converter(key, converted_data, errors, context)
-        return
-    except Invalid as e:
-        errors[key].append(e.error)
-        return
-    except TypeError as e:
-        # hack to make sure the type error was caused by the wrong
-        # number of arguments given.
-        if converter.__name__ not in str(e):
-            raise
-
+    signature = inspect.signature(converter)
+    nargs = len(signature.parameters)
+    if nargs == 1:
+        params = (converted_data.get(key),)
+    elif nargs == 2:
+        params = (converted_data.get(key), context)
+    elif nargs == 4:
+        params = (key, converted_data, errors, context)
+    else:
+        raise TypeError(
+            f"Wrong number of arguments(expected 1, 2 or 4): {nargs}")
```

Comment:
we should indicate which function had the wrong number of arguments because it won't be obvious from the traceback